dll/win32/shell32/CDefView.cpp

```diff
@@ -3326,3 +3338,71 @@ HRESULT WINAPI SHCreateShellFolderView(const SFV_CREATE *pcsfv,
     *ppsv = psv.Detach();
     return hRes;
 }
+
+/* CDefView callback interface */
+class CDefViewCB : public IShellFolderViewCB
```

Comment:
I'd suggest to use ATL, for example `CComObjectRootEx`:
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/shellext/zipfldr/CFolderViewCB.cpp;hb=e8d16d0a7d7c5adc70e40e66db15ed372647345b#l11